src/frontend/src/lib/types/token.ts

```diff
@@ -20,7 +21,16 @@ export interface TokenMetadata {
 	icon?: string;
 }
 
-export type RequiredToken = Required<Token>;
+export interface TokenAppearance {
+	displayName?: TokenDisplayName;
+}
+
+export type TokenDisplayName = {
```

Comment:
Let's rename this to emphasize that it's a custom choice made by the foundation. Without being to explicity, something like:

```
export type TokenCustomName = {
        prefix: string | undefined;
	customName: string;
};
```

or

```
export type TokenOisyName = {
        prefix: string | undefined;
	oisyName: string;
};
```

i.e. something that makes it obvious it's strictly for some mumbo jumbo

Reply:
i like the second one, it shows it is really specific to oisy